src/views/FacilityDetails.vue

```diff
@@ -545,6 +582,35 @@ export default defineComponent({
       });
       return externalMappingPopover.present()
     },
+    getDate(date: any) {
+      return DateTime.fromMillis(date).toFormat('dd LLL yyyy')
+    },
+    async removePartyFromFacility(party: any) {
+      try {
+        const resp = await FacilityService.removePartyFromFacility({
+          facilityId: party.facilityId,
+          fromDate: party.fromDate,
+          thruDate: DateTime.now().toMillis(),
+          partyId: party.partyId,
+          roleTypeId: party.roleTypeId
+        })
+
+        if(!hasError(resp)){
+          showToast("Party successfully removed from facility.")
+
+          // Refreshes the parties in facility
+          await this.store.dispatch('facility/getFacilityParties', { facilityId: this.facilityId })
+        } else {
+          throw resp
+        }
+      } catch(err) {
+        showToast("Failed to remove party from facility.")
+        logger.error(err)
+      }
+    },
+    getRoleTypeDesc(id: any) {
```

Comment:
I think we should store the roles in the state in the format `{ <roleType>: <desc>}`, so that we don't need to define this method and we can directly access roles desc, also for displaying the possible options for roleTypes in the add-party-modal we can iterate over the object directly.